url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/algorithms_and_systems_AI_Evolutionary/FuzzySystem.java

```diff
@@ -0,0 +1,938 @@
+// ##################################################################
+// ##################################################################
+// # FuzzySystem: Contains most of the common membership functions, # 
+// # hedges and defuzzification techniques used in fuzzy systems. 	#
+// # These are then used in methods to determine the genes for the 	#
+// # target chromosome to be used in the Genetic Algorithm.			#
+// ##################################################################
+// ##################################################################
+package algorithms_and_systems_AI_Evolutionary;
+
+import java.util.Arrays;
+import java.util.EnumSet;
+
+import expert_system.DesiredLoudness;
+import expert_system.Dynamic;
+import expert_system.DynamicShape;
+import expert_system.Feel;
+import expert_system.Key;
+import expert_system.Scale;
+import expert_system.Tonality;
+
+
+public class FuzzySystem {
+	
+	// ##########################################################
+	// #					   CONSTRUCTORS						#
+	// ##########################################################
+
+	public FuzzySystem() {
+		//BLANK
+	}
+	
+	// ##########################################################
+	// #					   FUNCTIONS						#
+	// ##########################################################
+	
+	
+	// ==========================================================
+	// =			       Membership Functions			    	=
+	// ==========================================================
+	
+	// Positive Grade function									
+	public double grade(double x, double x0, double x1){
+		
+		double fuzzied = 0;
+		
+		if(x <= x0)
+			fuzzied = 0;
+		else if(x > x0 && x < x1)
+			fuzzied = (x / (x1 - x0)) - (x0 / (x1 - x0));
+		else if(x >= x1)
+			fuzzied = 1;
+		
+		return fuzzied;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 45:54, 60:69.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.